app-backend/authentication/src/main/scala/Authentication.scala

```diff
@@ -70,6 +71,67 @@ trait Authentication extends Directives with LazyLogging {
   def getStringClaimOrBlank(claims: JWTClaimsSet, key: String): String =
     Option(claims.getStringClaim(key)).getOrElse("")
 
+  def getNameOrFallback(claims: JWTClaimsSet): String = {
+    val delegatedProfile = Option(claims.getJSONObjectClaim("delegatedProfile"))
+    val email = Option(claims.getStringClaim("email"))
+
+    val compareToEmail = (field: String) =>
+      (Option(claims.getStringClaim(field)), email) match {
+        case (fld @ Some(f), Some(e)) if f != e => fld
+        case (fld @ Some(_), None)              => fld
```

Comment:
You can collapse the bottom two cases here into `case (f @ _, _) => f`

Reply:
er wait, do I even need the `f @ _`? wouldn't just `(f, _) -> f` also work?